diff --git a/hw/net/Kconfig b/hw/net/Kconfig
index f9a1dfb80d..92136551ac 100644
--- a/hw/net/Kconfig
+++ b/hw/net/Kconfig
@@ -39,6 +39,11 @@ config E1000_PCI
     default y if PCI_DEVICES
     depends on PCI
 
+config ENA_PCI
+    bool
+    default y if PCI_DEVICES
+    depends on PCI
+
 config E1000E_PCI_EXPRESS
     bool
     default y if PCI_DEVICES || PCIE_DEVICES
diff --git a/hw/net/meson.build b/hw/net/meson.build
index 84f142df22..0de84ee6c1 100644
--- a/hw/net/meson.build
+++ b/hw/net/meson.build
@@ -8,6 +8,9 @@ system_ss.add(when: 'CONFIG_EEPRO100_PCI', if_true: files('eepro100.c'))
 system_ss.add(when: 'CONFIG_PCNET_PCI', if_true: files('pcnet-pci.c'))
 system_ss.add(when: 'CONFIG_PCNET_COMMON', if_true: files('pcnet.c'))
 system_ss.add(when: 'CONFIG_E1000_PCI', if_true: files('e1000.c', 'e1000x_common.c'))
+ena_dep = declare_dependency(compile_args: ['-I' + meson.current_source_dir() / '../../..'])
+system_ss.add(when: 'CONFIG_ENA_PCI', if_true: files('net_tx_pkt.c', 'net_rx_pkt.c'))
+system_ss.add(when: 'CONFIG_ENA_PCI', if_true: [files('../../../hw/ena.c', '../../../hw/ena_admin.c', '../../../hw/ena_tx.c', '../../../hw/ena_rx.c'), ena_dep])
 system_ss.add(when: 'CONFIG_E1000E_PCI_EXPRESS', if_true: files('net_tx_pkt.c', 'net_rx_pkt.c'))
 system_ss.add(when: 'CONFIG_E1000E_PCI_EXPRESS', if_true: files('e1000e.c', 'e1000e_core.c', 'e1000x_common.c'))
 system_ss.add(when: 'CONFIG_IGB_PCI_EXPRESS', if_true: files('net_tx_pkt.c', 'net_rx_pkt.c'))
diff --git a/tests/qtest/meson.build b/tests/qtest/meson.build
index 4897325d84..bbbb968b22 100644
--- a/tests/qtest/meson.build
+++ b/tests/qtest/meson.build
@@ -302,6 +302,15 @@ qos_test_ss.add(
   'adm1266-test.c',
   'ds1338-test.c',
   'e1000-test.c',
+  '../../../tests/ena_qos.c',
+  '../../../tests/ena_device_test.c',
+  '../../../tests/ena_admin_test.c',
+  '../../../tests/ena_aenq_test.c',
+  '../../../tests/ena_queue_test.c',
+  '../../../tests/ena_txpath_test.c',
+  '../../../tests/ena_llq_test.c',
+  '../../../tests/ena_rxpath_test.c',
+  '../../../tests/ena_offloads_test.c',
   'eepro100-test.c',
   'es1370-test.c',
   'lsm303dlhc-mag-test.c',
@@ -382,6 +391,8 @@ if get_option('replication').allowed()
   migration_colo_files = [files('migration/colo-tests.c')]
 endif
 
+ena_test_dep = declare_dependency(compile_args: ['-I' + meson.current_source_dir() / '../../..'])
+
 qtests = {
   'aspeed_hace-test': files('aspeed-hace-utils.c', 'aspeed_hace-test.c'),
   'aspeed_smc-test': files('aspeed-smc-utils.c', 'aspeed_smc-test.c'),
@@ -398,7 +409,7 @@ qtests = {
   'pxe-test': files('boot-sector.c'),
   'pnv-xive2-test': files('pnv-xive2-common.c', 'pnv-xive2-flush-sync.c',
                           'pnv-xive2-nvpg_bar.c'),
-  'qos-test': [chardev, io, qos_test_ss.apply({}).sources()],
+  'qos-test': [chardev, io, ena_test_dep, qos_test_ss.apply({}).sources()],
   'tpm-crb-swtpm-test': [io, tpmemu_files],
   'tpm-crb-test': [io, tpmemu_files],
   'tpm-tis-swtpm-test': [io, tpmemu_files, 'tpm-tis-util.c'],
